docs/SPEC.md

```diff
@@ -102,7 +102,7 @@ The following character sequences represent operators:
     -   <    !~   [   ]   ^
     *   >    =~   {   }   ?
     /   <=   =    ,   :   "
-    %   >=   <-   .   |>
+    %   >=   <-   .   |>  @
```

Comment:
I think it would probably be better if this wasn't an operator. It seems important to me that the `@` is a part of a specific metadata node. I wouldn't want `@ editions` to be a thing and I'd want us to require it to be `@editions`.

Reply:
I wonder if we should think more broadly about annotations and design something that is extensible for future use. The syntax could be something like

```
AnnotationStmt = "@" identifier expr
```
Or we could create a new kind of token if we don't want forbid a space in `@edition`. 

Allowing an arbitrary expression would allow us to be flexible in the future e.g., with an object expression
```
@some_annotation {
  foo: "bar",
  num: 100.3
}
```
For a simple case like like specifying an edition, we might want to use a string:
```
@edition "2022.1"
```
We wouldn't need to do it this way right away, but I think it would be smart to choose a syntax that could be part of a more general feature of the language in the future.

Reply:
Agreed

Reply:
I think I'd prefer to avoid this as the metadata system should be as simple as possible and allowing any possible expression means that the value for a module can be any expression. Once you start to open that, you start to open the question of what should be allowed (such as variables).

It might be good to have this be a string though. The primary reason is related to the scanner. In general, it's nicer if the scanner doesn't have to include parsing context. If we do `@edition 2022.1` we either have to create a new token for "edition pattern" or we have to use existing tokens. If we use existing tokens, then we're implementing the edition as a parser and whitespace will end up being ignored which would allow something like `@edition 2022 . 1`. I don't think that would be good. If we build it into the scanner, then the scanner either needs a special mode that it enters when it's parsing edition (we do this for regexes) in order for the parser to be able to recognize: `EditionsClause = EditionKey EditionValue`.

I kind of prefer the look of `@edition 2022.1` but it might not be a technically good idea unless we use some kind of preprocessor stage that causes them to be parsed differently.

We might want to consider that though. The `@` symbols are more like metadata similar to pragmas and we may just want them to be treated similar to comments by the parser rather than as part of the parser. Doing this will generally detach the system from the parser making it easier to affect the parser.